statsmodels/tsa2/sarimax/tests/test_tools.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+from statsmodels.tsa2.sarimax.tools import (
+    standardize_lag_order, validate_basic)
+
+import pytest
+from numpy.testing import assert_equal, assert_allclose, assert_raises
+
+
+def test_standardize_lag_order():
+    # Integer input
+    assert_equal(standardize_lag_order(0, title='test'), 0)
+    assert_equal(standardize_lag_order(3), 3)
+
+    # List input, lags
+    assert_equal(standardize_lag_order([]), 0)
+    assert_equal(standardize_lag_order([1, 2]), 2)
+    assert_equal(standardize_lag_order([1, 3]), [1, 3])
+
+    # Non-list iterable input, lags
+    assert_equal(standardize_lag_order((1, 2)), 2)
+    assert_equal(standardize_lag_order((1, 3)), [1, 3])
+
+    assert_equal(standardize_lag_order(np.array([1, 2])), 2)
+    assert_equal(standardize_lag_order(np.array([1, 3])), [1, 3])
+
+    # List input, booleans
+    print(standardize_lag_order([0]))
+    assert_equal(standardize_lag_order([0]), 0)
+    assert_equal(standardize_lag_order([1]), 1)
+    assert_equal(standardize_lag_order([0, 1]), [2])
+    assert_equal(standardize_lag_order([0, 1, 0, 1]), [2, 4])
+
+    # Non-list iterable input, lags
+    assert_equal(standardize_lag_order((0)), 0)
+    assert_equal(standardize_lag_order((1)), 1)
+    assert_equal(standardize_lag_order((0, 1)), [2])
+    assert_equal(standardize_lag_order((0, 1, 0, 1)), [2, 4])
+
+    assert_equal(standardize_lag_order(np.array([0])), 0)
+    assert_equal(standardize_lag_order(np.array([1])), 1)
+    assert_equal(standardize_lag_order(np.array([0, 1])), [2])
+    assert_equal(standardize_lag_order(np.array([0, 1, 0, 1])), [2, 4])
+
+    # Misc.
+    assert_equal(standardize_lag_order(np.array([[1], [3]])), [1, 3])
+
+    # Invalid input
+    assert_raises(TypeError, standardize_lag_order, None)
+    assert_raises(ValueError, standardize_lag_order, 1.2)
+    assert_raises(ValueError, standardize_lag_order, -1)
+    assert_raises(ValueError, standardize_lag_order,
+                  np.arange(4).reshape(2, 2))
+
+
+def test_validate_basic():
+    # Valid parameters
+    assert_equal(validate_basic([], 0, title='test'), [])
+    assert_equal(validate_basic(0, 1), [0])
+    assert_equal(validate_basic([0], 1), [0])
+    assert_equal(validate_basic(np.array([1.2, 0.5 + 1j]), 2),
+                 np.array([1.2, 0.5 + 1j]))
+    assert_equal(
+        validate_basic([np.nan, -np.inf, np.inf], 3, allow_infnan=True),
+        [np.nan, -np.inf, np.inf])
+
+    # Invalid parameters
+    assert_raises(ValueError, validate_basic, [], 1, title='test')
```

Comment:
pls use with `pytest.raises(ValueError, match="foo"):` syntax

Reply:
I know we differ on this point, but thanks for the comment.